tests/linked-list.spec.ts

```diff
@@ -206,6 +206,19 @@ describe('ARRAY LIKE USAGE', () => {
 	});
 });
 
+describe('Iteration', () => {
+	beforeEach(add10Items);
+	it('should not throw when using for of', () => {
```

Comment:
Testing with an arbitrary but consistent number of collection items is good! We need to go a little farther and test a few more cases. I

**The iterator's responsibilities:**
1. Maintain the current position within the collection during iteration.
2. Start at the correct item every time. The correct _starting item_ varies by ADT.
3. End at the correct item every time. The correct _ending item_ varies by ADT.
4. Move to the next element when requested.
5. Move to the previous element when requested.
6. Iterate items in the expected order. Again, _correct order_ varies by ADT.
7. next() should not throw if called after the iterator runs out of elements.
8. The return value for done and value should be correct.
 
For our iterator to be worth anything at all, the above tems must always be true. So, we need tests validating each responsibility is carried out correctly in all reasonable cases.

For example, what element does a freshly created iterator return before `next()` is called on an empty collection? A collection with 1 element? 10?

Some example tests based on the above:
- Does the return data correctly set done when getting the value on an empty array? Does it return correctly for 1 element collection? Check the expected value of done for collections with 0, 1, and n elements. For n, pick a small but reasonable number, e.g. dont pick 500 unless you're testing a boundary condition.
- Is done returned correctly when the iterator is beyond the end of the collection?
- Is value returned correctly when iterator is beyond end of collection?
- What happens to the iterator when an element is removed from collection?